Test/MultiProteaseParsimonyTest.cs

```diff
@@ -0,0 +1,179 @@
+                                break;
+                            }
+                            else { break; }
+
+
+
+                        case "EFGABC": psms.Add(new PeptideSpectralMatch(peptide.CompactPeptide(TerminusType.None), 0, 10, 0, scan, digestionParams)); break;
+
+
+                        case "XYZ": psms.Add(new PeptideSpectralMatch(peptide.CompactPeptide(TerminusType.None), 0, 10, 0, scan, digestionParams)); break;
+
+
+                        case "-XYZ-EFG": psms.Add(new PeptideSpectralMatch(peptide.CompactPeptide(TerminusType.None), 0, 10, 0, scan, digestionParams2)); break; 
+
+
+
+                    }
+                }
+            }
+
+            
+            List<ProductType> IonTypes = new List<ProductType>();
+            ProductType BnoB1ions = ProductType.BnoB1ions;
+            ProductType Yions = ProductType.Y;
+            IonTypes.Add(BnoB1ions);
+            IonTypes.Add(Yions);
+
+            List<DigestionParams> digestionParamsList = new List<DigestionParams>();
+            digestionParamsList.Add(digestionParams);
+            digestionParamsList.Add(digestionParams2);
+           // digestionParamsList.Add(digestionParams3);
+            ModificationMotif.TryGetMotif("M", out ModificationMotif motif1);
+            ModificationWithMass mod = new ModificationWithMass("Oxidation of M", "Common Variable", motif1, TerminusLocalization.Any, 15.99491461957);
+            List<ModificationWithMass> modVarList = new List<ModificationWithMass> { mod };
+
+            ModificationMotif.TryGetMotif("M", out ModificationMotif motif2);
+            ModificationWithMass mod2 = new ModificationWithMass("Oxidation of M", "Common Variable", motif2, TerminusLocalization.Any, 15.99491461957);
+            List<ModificationWithMass> modFixedList = new List<ModificationWithMass> { mod };
+            SequencesToActualProteinPeptidesEngine sequencesToActualProteinPeptidesEngine =
+            new SequencesToActualProteinPeptidesEngine(psms, p, modFixedList, modVarList, IonTypes, digestionParamsList, false, null);
+            var results = (SequencesToActualProteinPeptidesEngineResults)sequencesToActualProteinPeptidesEngine.Run();
+            var CompactPeptidesToProteinPeptidesMatching = results.CompactPeptideToProteinPeptideMatching;
+
+            ProteinParsimonyEngine ppe = new ProteinParsimonyEngine(CompactPeptidesToProteinPeptidesMatching, false, null);
+            var proteinAnalysisResults = (ProteinParsimonyResults)ppe.Run();
+          
+            List < ProteinGroup > proteinGroups = proteinAnalysisResults.ProteinGroups;
+            Assert.AreEqual(2, proteinGroups.Count);
+            Assert.AreEqual(2, proteinGroups.ElementAt(0).AllPeptides.Count);
+            Assert.AreEqual(1, proteinGroups.ElementAt(0).UniquePeptides.Count);
+            Assert.AreEqual(4, proteinGroups.ElementAt(1).AllPeptides.Count);
+            Assert.AreEqual(3, proteinGroups.ElementAt(1).UniquePeptides.Count);
+
```

Comment:
assert that the peptide sequences in AllPeptides and UniquePeptides are what you expect